src/ifcwrap/IfcParseWrapper.i

```diff
@@ -394,14 +394,21 @@ private:
 
 // The IfcFile* returned by open() is to be freed by SWIG/Python
 %newobject open;
+%newobject read;
 
 %inline %{
-	IfcParse::IfcFile* open(const std::string& s) {
+	IfcParse::IfcFile* open(const std::string& fn) {
 		IfcParse::IfcFile* f = new IfcParse::IfcFile();
-		f->Init(s);
+		f->Init(fn);
+		return f;
+	}
+    IfcParse::IfcFile* read(const std::string& data) {
+		IfcParse::IfcFile* f = new IfcParse::IfcFile();
+        std::string copiedData;
+        copiedData = data;
+		f->Init((void *)copiedData.c_str(), data.length());
```

Comment:
Hi, thanks for the updates, but this way of creating a copy is not sufficient. The issue with calling `std::string::c_str()` is that you loose all perks that come with C++ RAII or automatically garbage collected languages. Once the function `read()` is terminated `copiedData` will be freed. This means that the data pointed to by `c_str()` is also no longer available. Maybe it is not immediately apparent because after `IfcFile::Init()` the majority of data is parsed and present in memory. Strings and enumerations are parsed lazily and would probably crash in this approach. Additionally, once the IfcFile is freed it will try to clean up the `c_str()` which will not work, because it already has been freed or because it was allocated in a different way internally by the `std::string`. This is likely the error you report about. Long storey short, you really need to use the `memcpy()` approach used here: https://github.com/IfcOpenShell/IfcOpenShell/blob/master/src/ifcgeomserver/IfcGeomServer.cpp#L490